src/Tasks/Microsoft.NET.Build.Tasks/build/Microsoft.NET.Publish.targets

```diff
@@ -485,30 +485,36 @@ Copyright (c) .NET Foundation. All rights reserved.
                       PlatformLibraryName="$(MicrosoftNETPlatformLibrary)"
                       FilesToSkip="@(_ConflictPackageFiles);@(_PublishConflictPackageFiles)"
                       CompilerOptions="@(DependencyFileCompilerOptions)"
-                      PrivateAssetsPackageReferences="@(PrivateAssetsPackageReference)"
+                      ExcludeFromPublishPackageReferences="@(ExcludeFromPublishPackageReference)"
                       TargetManifestFileList="@(TargetManifestFileList)"
                       IsSelfContained="$(SelfContained)" />
 
   </Target>
 
   <!--
     ============================================================
-                                        ComputePrivateAssetsPackageReferences
+                                        ComputeExcludeFromPublishPackageReferences
 
-    Builds up the @(PrivateAssetsPackageReference) item by looking for @(PackageReference) items with 
+    Builds up the @(ExcludeFromPublishPackageReference) item by looking for @(PackageReference) items with 
     <PrivateAssets> metadata.
     ============================================================
     -->
   <PropertyGroup>
-    <ComputePrivateAssetsPackageReferences Condition="'$(ComputePrivateAssetsPackageReferences)' == ''">true</ComputePrivateAssetsPackageReferences>
+    <ComputeExcludeFromPublishPackageReferences Condition="'$(ComputeExcludeFromPublishPackageReferences)' == ''">true</ComputeExcludeFromPublishPackageReferences>
   </PropertyGroup>
 
-  <Target Name="ComputePrivateAssetsPackageReferences"
-          Condition="'$(ComputePrivateAssetsPackageReferences)' == 'true'">
+  <Target Name="ComputeExcludeFromPublishPackageReferences"
+          Condition="'$(ComputeExcludeFromPublishPackageReferences)' == 'true'">
 
     <ItemGroup>
-      <PrivateAssetsPackageReference Include="@(PackageReference)"
-                                     Condition="'%(PackageReference.PrivateAssets)' == 'All'" />
+      
+      <!-- PrivateAssets="All" means exclude from publish, unless ExcludeFromPublish metadata is specified separately -->
+      <PackageReference Update="@(PackageReference)" 
```

Comment:
The other option would be to put it in an `ItemGroup` in the .targets file but outside of any Target.  I thought that in this target was the right place for it to go, since this is the only place that metadata is consumed, and this keeps the behavior consistent if someone adds `PackageReference` items in a target, or after the import of the default .targets files.